packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -48,10 +48,12 @@ export const vModelText: ModelDirective<
     addEventListener(el, lazy ? 'change' : 'input', e => {
       if ((e.target as any).composing) return
       let domValue: string | number = el.value
-      if (trim) {
-        domValue = domValue.trim()
+      if (trim && castToNumber) {
```

Comment:
IMO, just remove `else` will be ok .
```suggestion
      if (trim ) {
      	 domValue = domValue.trim()
      }
      if(castToNumber){
      	domValue = toNumber(domValue)
      }
```

Reply:
thanks ! I think you are right👍.